src/app/modules/convai-checker/perspective-status.component.ts

```diff
@@ -1425,10 +1442,46 @@ export class PerspectiveStatus implements OnChanges, AfterViewInit, AfterViewChe
     });
   }
 
+  /** Wraps animations in Promises and chains them. */
+  private playAnimation(animation: Animation): void {
+    // Assign each animation a random id to keep track of it.
+    const id = Math.random();
+    this.pendingAnimations.push(id);
+
+    // Chain animations to complete after the previous one.
+    this.animationPromise = this.animationPromise.then(() => {
+      return this.getPlayAnimationPromise(animation, id);
```

Comment:
This does two things: 1) ensures the animations run sequentially (so I think that's a yes to your question) and 2) Prevents the event from getting emitted after the first animation has stopped, because we see that there is still one in the queue. The promise itself isn't used for anything. I originally tried listening directly on the Promise in the tests (e.g. await animationPromise) but the issue was that every time it was reassigned, the await would only wait for the Promise before it was reassigned.

Reply:
OK, so I think this is good for now, and we can have a chat about the semantics of when animations should end and how to best represent that later.